src/main/java/com/g4s8/ghman/user/UserDetails.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2019 Kirill Ch. (g4s8.publci@gmail.com)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to read
+ * the Software only. Permissions is hereby NOT GRANTED to use, copy, modify,
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.g4s8.ghman.user;
+
+import javax.json.Json;
+import javax.json.JsonObject;
+import javax.json.JsonObjectBuilder;
+import org.telegram.telegrambots.api.objects.Chat;
+
+/**
+ * Details about user.
+ * @since 1.0
+ */
+public interface UserDetails {
```

Comment:
@olenagerasimova can you add a `@todo` here to continue the work and have an implementation based on pg data so that `PgUser.telegram` can return a `UserDetails` instead of a `JsonObject`.

Reply:
@victornoel I do not see your point at all: telegram user data are stored as json object in db (check table `users` field `tname`). The main idea for this interface (for now) is to transform telegram user data given in any format into json object in order to store it in database. Why `PgUser.telegram()` should this return instance?

Reply:
@olenagerasimova as per the original todo that is solved here, the objective is that `User` abstraction should not be dependent on the fact that data is stored as json in the database.

Reply:
@victornoel but creating an abstraction that will expose user details from json has nothing to do with the interface `UserDetails` I created here, we'll need another abstraction for that

Reply:
@olenagerasimova hmmm, you are right, it doesn't work very well.

That's what I had in mind:
- `UserDetails` has methods for the each of the user details + a method `asJson()`
- User.telegram returns `UserDetails`
- We use `UserDetails` `FromChat` to store in the db
- We have another `UserDetails` `FromJson` to read from the db

It seems correct, but maybe not. What should we do instead?

Reply:
@victornoel I think we leave `UserDetails` and its implementation as is and simply create separate abstraction that exposes every field of this json object and it's implementation `FromJson`, than we change return type of `User.telegram()` to this new abstraction

Reply:
@olenagerasimova in that case there would be the need for a test to ensure that this `FromJson` implementation is actually working with the `UserDetails.FromChat` implementation result. And `UserDetails` should then be renamed something like `PgUserDetails` to be clear that it is strongly related to the serialization of data inside the database.

@g4s8 can you chip in and tell us what would be best to do, architecturally-wise?

Reply:
@g4s8 can we get some ruling on this so that things can go forward?

Reply:
@victornoel I do not mind to rename into `UserDetails` into `PgUserDetails` and speaking about test to check that this not existing yet `FromJson` implementation is actually working with the `UserDetails.FromChat` implementation result I think that would be up to `PgUsers` integration test

Reply:
@victornoel @olenagerasimova why do we need this class? Where it will be used? I'd start introducing new abstractions on-demand only, when we start using these user-details

Reply:
@g4s8 not to be a drag, but maybe it's best if we don't let new todos enter the codebase if they go against some architectural choices next time?
Part of the rationale behind the todo currently being solved by this PR is explained in the comment above: https://github.com/g4s8/ghman/pull/72#discussion_r304617241 and also discussed when it was introduced after #18.

Reply:
@victornoel you're right, it's my fault (I didn't read all PR puzzles entirely) and PR reviewer too. The concept of todo puzzles it that it should not extend current scope, performer may submit only incomplete job as puzzles, not new features and improvements. Performer may submit new ticket if he/she wants to propose new feature or improvement.

Reply:
@g4s8 yep, so I'm at fault too since I participated in those puzzle introductions.
@olenagerasimova and @g4s8 I propose to just remove the puzzle and do nothing else in this PR then. We will see if more abstractions are needed later, but for now those tests and public classes/interface are too much then. WDYT?

Reply:
@victornoel todo removed

Reply:
@olenagerasimova I meant remove all the changes from this PR and just keep the todo removal from #53. Apparently the original todo had no reason to exist.

Reply:
@victornoel hm, but we'd still like to test `JsonUserDetails`, don't we? May be I can remove the abstraction and leave the test?

Reply:
@olenagerasimova well, since as @g4s8 said, we don't even know how it is used, there is nothing to test in terms of object behaviour, it was my mistake to go too fast into design of a future feature when I proposed the original todo that was asking for tests. I think it's good the code is isolated in an object instead of a static method like it was before, but it is still part of the inside of `PgUsers` in the end.
But let's wait for @g4s8 guidance on this then.

Reply:
@g4s8 ping